rpc/wrtc_server_stream.go

```diff
@@ -293,12 +293,22 @@ func (s *webrtcServerStream) processHeaders(headers *webrtcpb.RequestHeaders) {
 	}
 
 	s.headersReceived = true
-	s.ch.server.activeBackgroundWorkers.Add(1)
+	s.ch.server.processHeadersLock.Lock()
+	// Check once more if underlying webrtcServer was `Stop`ped before lock was
+	// acquired (context was canceled).
```

Comment:
This is new logic: if we `Lock()` and the underlying server was closed (context was canceled), we can `closeWithSendError` and `return` as is currently done on L283. I'm not exactly sure what the behavior is here when peer connections are closed, but it seems like we were already fine calling `closeWithSendError` after the server's context had expired...